aten/src/ATen/ExpandUtils.h

```diff
@@ -9,6 +9,14 @@ namespace at {
 std::vector<int64_t> infer_size(IntList a, IntList b);
 std::tuple<std::vector<int64_t>, std::vector<int64_t> > inferExpandGeometry(const Tensor &tensor, IntList sizes);
 
+inline void check_defined(std::vector<Tensor> tensors, const std::string &api_name) {
```

Comment:
This will allocate a `std::vector` every call. Use `std::initializer_list` instead with either a `Tensor*` or `std::reference_wrapper<Tensor>` to avoid the refcounting.

Reply:
TensorList should avoid the refcounting, right? (and be more ATen-like).